best_practices.md

```diff
@@ -0,0 +1,195 @@
+# Best practices for programming diagnostics
+
+To facilitate code integration, code sharing between developers, reusability, easy
+maintenance, and readability, it is recommended that these recommendations
+are followed when developing new diagnostic scripts.
+
+## Make your code open-source
+
+It is recommended that you make your code
+[open-source](https://the-turing-way.netlify.app/reproducible-research/open/open-source.html)
+from the start, to facilitate reproducible science and encourage potential
+[collaboration](https://the-turing-way.netlify.app/collaboration/collaboration.html).
+If you have reservations about making your code open source before publication,
+consider developing it in a private [git repository](#use-git-for-version-control) and making it
+public once you have submitted your paper.
+
+## Add a license
+
+It is critical that you share your code with a
+[license](https://the-turing-way.netlify.app/reproducible-research/licensing.html),
+because others will not legally be allowed to use it if it does not have a license.
+Make sure that any [dependencies](#use-libraries) you use or plan to use are
+[compatible](https://the-turing-way.netlify.app/reproducible-research/licensing/licensing-compatibility.html)
+with the license you choose.
+
+## Use git for version control
+
+It is recommended that you make use of a
+[version control](https://the-turing-way.netlify.app/reproducible-research/vcs.html)
+tool.
+In particular, it is recommended that you use git, because of all available
+version control systems it best facilitates (future) collaboration.
+
+## Make your code citable
+
+In order to get scientific credit for your work, you may wish to make it
+[easy to cite your code](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files).
+Many papers ask for a [DOI](https://www.doi.org/) for your code these days.
+A convenient way to obtain that is by linking your
+[GitHub](#host-your-code-on-a-website-with-an-issue-tracker) repository with
+[Zenodo](https://zenodo.org/) as described in
+[Making your code citable](https://guides.github.com/activities/citable-code/).
+
+## Choose a suitable programming language
```

Comment:
Just a comment ... I have heard people like also Julia and there is still R. Not sure if it is still supported by esmvaltool.

Reply:
Yes, these are supported by the ESMValTool, but I honestly cannot recommend them for doing climate science because as far as I am aware, essential libraries for handling climate data are missing in these ecosystems.

For example, all R diagnostics that we have in the ESMValTool use CDO command-line instructions for processing climate data which is error-prone and computationally inefficient.

For the Julia diagnostics, the library situation is even worse because not only are there no libraries that can deal with climate data (beyond the most basic read a NetCDF file into an in-memory array), for plotting it depends on using Python's matplotlib which is causing all kinds of installation headaches. 

Or would you have some good libraries to suggest for these languages?

Reply:
In R we have a specific niche that is (or at least was 2-3 years ago) much better covered in R, like the verification of climate predictions: https://cran.r-project.org/web/packages/SpecsVerification/index.html

Reply:
Good point, I added your text on this topic